consensus/istanbul/backend/announce.go

```diff
@@ -618,144 +621,216 @@ func (sb *Backend) validateAnnounce(announceData *announceData) (bool, error) {
 	return true, nil
 }
 
-// regossipAnnounce will regossip a received announce message.
-// If this node regossiped an announce from the same source address within the last 5 minutes, then it wouldn't regossip.
-// This is to prevent a malicious validator from DOS'ing the network with very frequent announce messages.
-// Note that even if the validator is not malicious, but changing their enode very frequently, the
-// other validators will eventually get that validator's latest enode, since all nodes will periodically check it's neighbors
-// for updated announce messages.
-func (sb *Backend) regossipAnnounce(msg *istanbul.Message, payload []byte) error {
-	logger := sb.logger.New("func", "regossipAnnounce", "announceSourceAddress", msg.Address)
+// scheduleAnnounceRegossip schedules a regossip for an announce message after
+// its cooldown period. If one is already scheduled, it's overwritten.
+func (sb *Backend) scheduleAnnounceRegossip(msg *istanbul.Message, msgTimestamp uint, payload []byte) error {
+	logger := sb.logger.New("func", "scheduleAnnounceRegossip", "msg address", msg.Address, "msg timestamp", msgTimestamp)
+	sb.scheduledAnnounceRegossipsMu.Lock()
+	defer sb.scheduledAnnounceRegossipsMu.Unlock()
+
+	// If another regossip has been scheduled for the message address already, cancel it
+	if scheduledRegossip := sb.scheduledAnnounceRegossips[msg.Address]; scheduledRegossip != nil {
+		scheduledRegossip.Timer.Stop()
+	}
 
 	sb.lastAnnounceGossipedMu.RLock()
-	if lastGossipTs, ok := sb.lastAnnounceGossiped[msg.Address]; ok {
-		if time.Since(lastGossipTs) < 5*time.Minute {
-			logger.Trace("Already regossiped msg from this source address within the last 5 minutes, so not regossiping.")
-			sb.lastAnnounceGossipedMu.RUnlock()
+	lastGossip := sb.lastAnnounceGossiped[msg.Address]
+	if lastGossip == nil {
+		logger.Debug("Last announce gossiped not found")
+		sb.lastAnnounceGossipedMu.RUnlock()
+		return nil
+	}
+	scheduledTime := lastGossip.Time.Add(announceGossipCooldownDuration)
+	sb.lastAnnounceGossipedMu.RUnlock()
+
+	duration := time.Until(scheduledTime)
+	// If by this time the cooldown period has ended, just regossip
+	if duration < 0 {
```

Comment:
Nit: Might as well be `<=`
```suggestion
	if duration <= 0 {
```